pkg/kf/apps/push.go

```diff
@@ -156,13 +175,40 @@ func setupRoutes(cfg pushConfig, appName string, r []v1alpha1.RouteSpecFields) (
 	}
 }
 
+func noCfgScaling(cfg pushConfig) bool {
+	return cfg.MinScale == nil && cfg.MaxScale == nil
+}
+
+func noScaling(app *v1alpha1.App) bool {
+	return app.Spec.Instances.Exactly == nil &&
+		app.Spec.Instances.Min == nil &&
+		app.Spec.Instances.Max == nil
+}
+
 func mergeApps(cfg pushConfig, hasDefaultRoutes bool) func(newapp, oldapp *v1alpha1.App) *v1alpha1.App {
 	return func(newapp, oldapp *v1alpha1.App) *v1alpha1.App {
 
 		if len(oldapp.Spec.Routes) > 0 && hasDefaultRoutes {
 			newapp.Spec.Routes = oldapp.Spec.Routes
 		}
 
+		// Scaling overrides
+		if noCfgScaling(cfg) {
+			// Looks like the user did not set a new value, use the old one
+			newapp.Spec.Instances.Exactly = oldapp.Spec.Instances.Exactly
+			newapp.Spec.Instances.Min = oldapp.Spec.Instances.Min
+			newapp.Spec.Instances.Max = oldapp.Spec.Instances.Max
+		}
+
+		// Default scaling
+		if noCfgScaling(cfg) && noScaling(oldapp) {
+			// No scaling in old or new, go with a default of 1. This is to
+			// match expectaions for CF users. See
+			// https://github.com/google/kf/issues/8 for more context.
+			singleInstance := 1
```

Comment:
Should this be done server-side as part of the defaulting webhook?

Reply:
I don't think so. For example, what if they set `-i=0`? How would the webhook know?